src/words.rs

```diff
@@ -1,7 +1,9 @@
 use std::str;
 use rand::Rng;
 
-pub trait WordGenerator: Clone {
+
+
+pub trait WordGenerator: Clone + Send + 'static {
```

Comment:
This is actually most of the work. By adding `Send` we're telling the compiler it must ensure that it's safe to send this trait between threads. By adding `'static`, we're telling it that this trait may only have static references or owned types, i.e. the lifetime of this trait is not affected by any other lifetimes.